### PR TITLE
Feature/usage hints

### LINKED ISSUE #385: Add hint for mouse interaction
- [x] Add hints for mouse interaction (how to rotate, how to move camera, how to zoom (?) ) to lower right corner
- [x] Make hints disappear if user uses rotation / moving / ...

2015-03-26 mouse hints

### PR DESCRIPTION
**Pull request by AtiX**
closes #385

### CODE REVIEW

**Comment by stuikomma** [comment]
The zoom hint will only disappear when clicking the scroll wheel. But you can also use the wheel without clicking.

**Comment by stuikomma** [comment]
Maybe we want to show the hints only to first-time users?

File: public/styles/app/index.styl [hunk 1]
```diff
@@ -37,3 +37,14 @@ h1 a
...
+#usageHintContainer
+	position absolute
+	bottom 0
+	right 0
+	margin-bottom 10px
+	margin-right 10px
+	max-width 100px
+	
+	img
+		width 100px
```
Comment (Steditor) [inline public/styles/app/index.styl]: no newline at eof

**Comment by stuikomma** [comment]
At least for me the first two icons should be swapped because I would rather first have a look at my model from different angles before brushing.

**Comment by AtiX** [comment]
@stuikomma  :>

File: src/client/ui/HintUi.coffee [hunk 2]
```diff
@@ -0,0 +1,53 @@
+class HintUi
+	constructor: ->
+		@hintContainer = $('#usageHintContainer')
```
Comment (Steditor) [inline src/client/ui/HintUi.coffee]: https://github.com/style-guides/CoffeeScript#jquery

File: views/app/app.jade [hunk 3]
```diff
@@ -25,6 +25,12 @@ block body
 				= ' '
 				a(href='http://get.webgl.org') get.webgl.org
 
+		#usageHintContainer
+			img(src='img/move-01.svg')#moveHint
```
Comment (Steditor) [inline views/app/app.jade]: Why are the files named `-01`?
I would also suggest putting them in a sub-folder.

File: views/app/app.jade [hunk 4]
```diff
@@ -25,6 +25,12 @@ block body
...
+			img(src='img/zoom-01.svg')#zoomHint
```
Comment (Steditor) [inline views/app/app.jade]: Maybe use a magnifier? I don't think the arrow conveys the zooming functionality.

File: views/app/app.jade [hunk 5]
```diff
@@ -25,6 +25,12 @@ block body
...
+			img(src='img/rotate-01.svg')#rotateHint
```
Comment (Steditor) [inline views/app/app.jade]: As above, maybe make it 3D: https://cloud.githubusercontent.com/assets/236678/6843858/f3a28560-d3a6-11e4-96eb-3acd6212f332.jpg

**Comment by stuikomma** [comment]
All usage hints are disabled when revisiting the page. It should restore the previous state.

**Comment by stuikomma** [comment]
sometimes brushing can cause the rotate hint to disappear

- [ ] fixed

**Comment by stuikomma** [comment]
Better reverse the default behaviour, i.e. hide hints by default and show if needed. (Otherwise they will always be visible until JS loaded)

- [ ] done

**Comment by stuikomma** [comment]
For our own benefit we might want to make FPS + usage hints look nicer.

- [ ] done (optional!)

**Comment by Steditor** [comment]
In the long run: Swap to https://github.com/brickify/three-pointer-controls/issues/26

**Comment by AtiX** [comment]
:rocket:

**Pull request merged**